src/elements/content-sidebar/activity-feed/Avatar.js

```diff
@@ -14,46 +14,46 @@ type Props = {
 
 type State = {
     avatarUrl?: ?string,
+    isPending: boolean,
 };
 
+// Note: setting a key is required, re-render with a new user is not implemented
+
 class Avatar extends React.PureComponent<Props, State> {
-    state = {
-        avatarUrl: null,
-    };
-
-    /**
-     * Success handler for getting avatar url
-     *
-     * @param {string} avatarUrl the user avatar url
-     */
-    getAvatarUrlHandler = (avatarUrl: ?string) => {
-        this.setState({
+    constructor(props: Props) {
+        super(props);
+        // short-circuit promises by checking if we already have a url
+        // or if we don't have a method to get one
+        const avatarUrl = props.user.avatar_url || null;
+        this.state = {
             avatarUrl,
-        });
-    };
-
-    /**
-     * Gets the avatar URL for the user from the getAvatarUrl prop
-     *
-     * @return {Promise<?string>} Promise which resolve with the avatar url string
-     */
-    getAvatarUrl(): Promise<?string> {
-        const { user, getAvatarUrl }: Props = this.props;
-
-        const avatarPromise = getAvatarUrl ? getAvatarUrl(user.id) : Promise.resolve(user.avatar_url);
-        return avatarPromise.then(this.getAvatarUrlHandler);
+            // pending state means we are using getAvatarUrl to get a url asynchronously
+            isPending: !!props.getAvatarUrl && !avatarUrl,
+        };
     }
 
     componentDidMount() {
         this.getAvatarUrl();
     }
 
+    async getAvatarUrl() {
+        const { user, getAvatarUrl } = this.props;
+        const { isPending } = this.state;
+        // don't fetch if url exists or no handler was passed
+        if (!isPending || !getAvatarUrl) return;
```

Comment:
do you mind expanding this out with brackets? We don't really do this anywhere else

Reply:
Ok, but as a followup please create a lint rule for this if you care about it.